README.md

```diff
@@ -1,135 +1,287 @@
 
-To use `Application Default Credentials`, You first need to download a set of
-JSON credentials for your project. Go to **APIs & Auth** > **Credentials** in
-the [Google Developers Console][devconsole] and select
-**Service account** from the **Add credentials** dropdown.
+To use `Application Default Credentials`, You first need to download a set of JSON credentials for your project. Go to **APIs & Auth** > **Credentials** in the [Google Developers Console][devconsole] and select **Service account** from the **Add credentials** dropdown.
 
 > This file is your *only copy* of these credentials. It should never be
 > committed with your source code, and should be stored securely.
 
-Once downloaded, store the path to this file in the
-`GOOGLE_APPLICATION_CREDENTIALS` environment variable.
+Once downloaded, store the path to this file in the `GOOGLE_APPLICATION_CREDENTIALS` environment variable.
 
 #### Enable the API you want to use
 
-Before making your API call, you must be sure the API you're calling has been
-enabled. Go to **APIs & Auth** > **APIs** in the
-[Google Developers Console][devconsole] and enable the APIs you'd like to
-call. For the example below, you must enable the `DNS API`.
+Before making your API call, you must be sure the API you're calling has been enabled. Go to **APIs & Auth** > **APIs** in the [Google Developers Console][devconsole] and enable the APIs you'd like to call. For the example below, you must enable the `DNS API`.
+
+
+#### Choosing the correct credential type automatically
 
-#### Call an API
+Rather than manually creating an OAuth2 client, JWT client, or Compute client, the auth library can create the correct credential type for you, depending upon the environment your code is running under.
 
-As long as you update the environment variable below to point to *your* JSON
-credentials file, and the fill in the placeholder variables from your project,
-the following snippet should work.
+For example, a JWT auth client will be created when your code is running on your local developer machine, and a Compute client will be created when the same code is running on Google Cloud Platform.
+
+The code below shows how to retrieve a default credential type, depending upon the runtime environment. The createScopedRequired must be called to determine when you need to pass in the scopes manually, and when they have been set for you automatically based on the configured runtime environment.
 
 ```js
-var google = require('googleapis');
-var GoogleAuth = require('google-auth-library').GoogleAuth;
+const GoogleAuth = require('google-auth-library').GoogleAuth;
+const auth = new GoogleAuth();
+
+/**
+ * Acquire a client, and make a request to an API that's enabled by default.
+ */
+async function main() {
+  try {
+    const adc = await getADC();
+    const url = `https://www.googleapis.com/dns/v1/projects/${adc.projectId}`;
+    const res = await adc.client.request({url});
```

Comment:
I don't have strong feelings here, but it would be preferable if we downplayed `client.request` in the examples. We really should not be encouraging the use of the auth library as a http client library. Ideally users use their own `request` or other client library.

This is definitely the most succinct example you can show and it seems convenient, but I don't think it scales when the needs of the user grow beyond what `client.request` can offer (e.g. multipart support). I've seen this questions about this show up many times.

A compromise might be if we added a section on how to use `authorizeRequest` with a user provided http client library.

Reply:
As it stands today, this is the most common usage of the API, isn't it?  We specifically use this in the Google API Node.js Client.  I mean I hear what you're saying, but it just doesn't reflect the reality of the API as it's structured and used today.  

Lets open another issue for a future version (v2?) that cleans this up to avoid bundling the request client lib with the package. Thoughts?

Reply:
I know that the Google API Node.js client does this, but I wasn't able to find other examples. Is it really the most common usage?

Reply:
Yeah, I did some spot checking and this stuff was all over the map.  For something like ADC - I'm not even sure how to show someone to use that without using the request object *scratches head*.  Maybe for the individual clients (JWT, OAuth2, Compute) it may make sense to show different examples.  To be honest - I'm very open to adding more samples and tweaking this stuff, but I do feel like this PR is burning a hole in my pocket :)  I'd like to get this branch merged into master soonish.  What do you think?

Reply:
Since I already indicated that I don't have strong feelings, I am not going to block on this.